service/frontend/adminHandler.go

```diff
@@ -227,6 +227,68 @@ func (adh *AdminHandler) AddSearchAttributes(ctx context.Context, request *admin
 	}, nil
 }
 
+// RemoveSearchAttributes add search attribute to the cluster.
+func (adh *AdminHandler) RemoveSearchAttributes(ctx context.Context, request *adminservice.RemoveSearchAttributesRequest) (_ *adminservice.RemoveSearchAttributesResponse, retError error) {
+	defer log.CapturePanic(adh.GetLogger(), &retError)
+
+	scope, sw := adh.startRequestProfile(metrics.AdminRemoveSearchAttributesScope)
+	defer sw.Stop()
+
+	// validate request
+	if request == nil {
+		return nil, adh.error(errRequestNotSet, scope)
+	}
+
+	if len(request.GetSearchAttributes()) == 0 {
+		return nil, adh.error(errSearchAttributesNotSet, scope)
+	}
+
+	indexName := request.GetIndexName()
+	if indexName == "" {
+		indexName = adh.ESConfig.GetVisibilityIndex()
+	}
+
+	currentSearchAttributes, err := adh.Resource.GetSearchAttributesProvider().GetSearchAttributes(indexName, true)
+	if err != nil {
+		return nil, adh.error(serviceerror.NewInternal(fmt.Sprintf(errUnableToGetSearchAttributesMessage, err)), scope)
+	}
+
+	newCustomSearchAttributes := map[string]enumspb.IndexedValueType{}
+	for saName, saType := range currentSearchAttributes.Custom() {
+		newCustomSearchAttributes[saName] = saType
+	}
+
+	for _, saName := range request.GetSearchAttributes() {
+		if !currentSearchAttributes.IsDefined(saName) {
+			return nil, adh.error(serviceerror.NewInvalidArgument(fmt.Sprintf(errSearchAttributeDoesntExistMessage, saName)), scope)
+		}
+		if _, ok := newCustomSearchAttributes[saName]; !ok {
+			return nil, adh.error(serviceerror.NewInvalidArgument(fmt.Sprintf(errUnableToRemoveNonCustomSearchAttributesMessage, saName)), scope)
+		}
+		delete(newCustomSearchAttributes, saName)
+	}
+
+	err = adh.Resource.GetSearchAttributesManager().SaveSearchAttributes(indexName, newCustomSearchAttributes)
+	if err != nil {
+		return nil, adh.error(serviceerror.NewInternal(fmt.Sprintf(errUnableToSaveSearchAttributesMessage, err)), scope)
+	}
+
+	// Get current state and return it.
+	resp, err := adh.getSearchAttributes(ctx, indexName, "")
+	if err != nil {
+		// Don't return error in case of getSearchAttributes error. Just log it.
+		adh.GetLogger().Error("Unable to get search attributes back while adding them.", tag.Error(err))
+		metricsGetSearchAttributesScope := adh.GetMetricsClient().Scope(metrics.AdminGetSearchAttributesScope)
+		metricsGetSearchAttributesScope.IncCounter(metrics.ServiceFailures)
+	}
+
+	return &adminservice.RemoveSearchAttributesResponse{
+		CustomAttributes: resp.GetCustomAttributes(),
+		SystemAttributes: resp.GetSystemAttributes(),
+		Mapping:          resp.GetMapping(),
+	}, nil
+}
```

Comment:
```
// Don't return error in case of getSearchAttributes error. Just log it.
```

resp will contain all nil-s?

Reply:
API should not return garbage data

Reply:
`Remove...` is to remove, not to get. Returning an error here, will give user (`tctl`) an impression that field wasn't removed. But at this point it was successfully removed but something else failed. User can always call `Get...` one more time and got actual error message. I can return simple empty response here instead of detailed info and just call `Get...` separately from `tctl`. `Add...` needs to be modified too then.

Reply:
API in no case can return garbage data

```
Remove... is to remove, not to get
```
if ^ is the case, then remove API should return nothing by design, let user / CLI call server again for the latest info

Reply:
It is not garbage. It is just incomplete data. But I get your point and changed `RemoveSearchAttributes` API to return empty response in case of success. I will change `AddSearchAttributes` with another PR (it is compatible change).